lib/membrane_http_adaptive_stream/sink_bin.ex

```diff
@@ -186,10 +198,16 @@ defmodule Membrane.HTTPAdaptiveStream.SinkBin do
   end
 
   @impl true
-  def handle_element_end_of_stream({:sink, _}, _ctx, state) do
+  def handle_element_end_of_stream({:sink, _}, _ctx, %{streams_counter: 1} = state) do
     {{:ok, notify: :end_of_stream}, state}
```

Comment:
I would decrese `streams_counter` anyway

```suggestion
    state = Map.update!(state, :streams_counter, &(&1 - 1))
    {{:ok, notify: :end_of_stream}, state}
```